src/msw/printdlg.cpp

```diff
@@ -964,6 +984,14 @@ bool wxWindowsPrintDialog::ConvertFromNative( wxPrintDialogData &data )
     {
         data.SetFromPage(pd->lpPageRanges[0].nFromPage);
         data.SetToPage(pd->lpPageRanges[0].nToPage);
+
+        wxVector<wxPrintPageRange>& ranges = data.GetPageRanges();
+        ranges.clear();
```

Comment:
This is a bit of a weird pattern, shouldn't we just have `SetPageRanges()` instead?

Reply:
You could, but actually it is not necessary to create a new vector beforehand....

Reply:
Is this an efficiency concern? Because I really don't think temporarily allocating extra 1KiB of memory is a problem.

Reply:
I'm thinking more of flexibility then on efficiency ..

Reply:
Well, using (const) `Get` + `Set` is exactly as flexible, as you just can make a copy of the vector and then it's the same as using the current (non-const) `Get`. The only difference is efficiency (but I don't think it matters here) and consistency with the rest of the API -- which I think does matter and wx API almost universally uses const `Get` and `Set` instead of allowing to modify things in place (there are some exceptions, of course...).